sensorlib/src/noise_models/AbstractNoiseModel.py

```diff
@@ -53,3 +53,45 @@ def apply_list_inclusion_noise(self, object_list):
         :return: List of DetectedObject objects with noise applied.
         """
         pass
+
+    @abstractmethod
+    def apply_position_covariance(self, object_list):
+        """
+        Apply noise related to position covariance to the object positions.
+
+        :param object_list: List of DetectedObject objects to apply noise to.
+        :return: List of DetectedObject objects with noise applied.
+        """
+        pass
+
+    @abstractmethod
+    def apply_orientation_covariance(self, object_list):
+        """
+        Apply noise related to orientation covariance to the object orientations.
+
+        :param object_list: List of DetectedObject objects to apply noise to.
+        :return: List of DetectedObject objects with noise applied.
+        """
+        pass
+
+    @abstractmethod
+    def apply_velocity_covariance(self, object_list):
+        """
+        Apply noise related to velocity covariance to the object velocities.
+
+        :param object_list: List of DetectedObject objects to apply noise to.
+        :return: List of DetectedObject objects with noise applied.
+        """
+        pass
+
+    @abstractmethod
+    def apply_angular_velocity_covariance(self, object_list):
```

Comment:
Should be renamed to `apply_angular_velocity_covariance_noise` to match the naming scheme of the other functions.